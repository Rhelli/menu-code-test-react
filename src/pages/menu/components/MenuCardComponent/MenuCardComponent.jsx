import React from 'react';
import PropTypes from 'prop-types';
import CourseMapperComponent from '../../../../common/courseMapperComponent/CourseMapperComponent.jsx';
import styles from './MenuCardComponent.module.scss';

const menuData = require('../../../../../menu-data.json');

const MenuCardComponent = ({ submitOrderAddition }) => {
  const { starters, mains, desserts } = menuData;

  return (
    <div className={styles.menuCardContainer}>
      <CourseMapperComponent
        foodArray={starters}
        onClickFunc={submitOrderAddition}
        course='starters'
      />
      <CourseMapperComponent
        foodArray={mains}
        onClickFunc={submitOrderAddition}
        course='mains'
      />
      <CourseMapperComponent
        foodArray={desserts}
        onClickFunc={submitOrderAddition}
        course='desserts'
      />
    </div>
  );
};

MenuCardComponent.propTypes = {
  submitOrderAddition: PropTypes.func.isRequired
};

export default MenuCardComponent;
