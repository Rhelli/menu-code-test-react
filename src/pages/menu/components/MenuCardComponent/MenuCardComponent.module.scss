.menuCardContainer {
  display: flex;
  flex-flow: column nowrap;
  width: 100%;
  height: 100%;
  border-radius: 3px;
  border: 2px solid #707070;

  h1 {
    width: 100%;
    font-size: 30px;
    line-height: 100%;
    padding: 20px 20px;
    border-bottom: 1px solid #d8d9db;
  }
}