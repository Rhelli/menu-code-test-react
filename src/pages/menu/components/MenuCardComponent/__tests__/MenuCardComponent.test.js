import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import MenuCardComponent from '../MenuCardComponent';

Enzyme.configure({ adapter: new Adapter() });

const submitOrderAddition = jest.fn(() => 'test');
const menuCardComponent = (<MenuCardComponent submitOrderAddition={submitOrderAddition} />);

describe('The appearance and functionality of the MenuCardComponent', () => {
  const container = mount(<MenuCardComponent submitOrderAddition={submitOrderAddition} />);
  it('Matches the previous snapshot of the component', () => {
    const tree = renderer.create(menuCardComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Maps and displays selectable starters taken from the menu data', () => {
    expect(container.find('.startersMenu').childAt(1).children()).toHaveLength(4);
    expect(container.find('.startersMenu').childAt(1).childAt(1).props('onClick')).toBeTruthy();
  });

  it('Displays the name and price of the starters it maps', () => {
    expect(container.find('.startersMenu').childAt(1).childAt(1).childAt(0).text()).toBe('Pâté');
    expect(container.find('.startersMenu').childAt(1).childAt(1).childAt(1).text()).toBe('£5.00');
  });

  it('Maps and displays selectable mains taken from the menu data', () => {
    expect(container.find('.mainsMenu').childAt(1).children()).toHaveLength(4);
    expect(container.find('.mainsMenu').childAt(1).childAt(1).props('onClick')).toBeTruthy();
  });

  it('Displays the name and the price of the mains it maps', () => {
    expect(container.find('.mainsMenu').childAt(1).childAt(1).childAt(0).text()).toBe('Meatballs');
    expect(container.find('.mainsMenu').childAt(1).childAt(1).childAt(1).text()).toBe('£11.50');
  });

  it('Maps and displays selectable desserts taken from the menu data', () => {
    expect(container.find('.dessertsMenu').childAt(1).children()).toHaveLength(4);
    expect(container.find('.dessertsMenu').childAt(1).childAt(1).props('onClick')).toBeTruthy();
  })
});