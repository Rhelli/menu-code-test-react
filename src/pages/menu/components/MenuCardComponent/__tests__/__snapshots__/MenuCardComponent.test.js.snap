// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance and functionality of the MenuCardComponent Matches the previous snapshot of the component 1`] = `
<div
  className="menuCardContainer"
>
  <div
    className="courseMenu"
  >
    <h3>
      Starters
    </h3>
    <div
      className="foodOptionBar"
      onClick={[Function]}
    >
      <p
        className="foodName"
      >
        Soup
      </p>
      <p
        className="foodPrice"
      >
        £3.00
      </p>
    </div>
    <div
      className="foodOptionBar"
      onClick={[Function]}
    >
      <p
        className="foodName"
      >
        Pâté
      </p>
      <p
        className="foodPrice"
      >
        £5.00
      </p>
    </div>
    <div
      className="foodOptionBar"
      onClick={[Function]}
    >
      <p
        className="foodName"
      >
        Bruschetta
      </p>
      <p
        className="foodPrice"
      >
        £4.50
      </p>
    </div>
    <div
      className="foodOptionBar"
      onClick={[Function]}
    >
      <p
        className="foodName"
      >
        Prawn cocktail
      </p>
      <p
        className="foodPrice"
      >
        £6.00
      </p>
    </div>
  </div>
  <div
    className="courseMenu"
  >
    <h3>
      Mains
    </h3>
    <div
      className="foodOptionBar"
      onClick={[Function]}
    >
      <p
        className="foodName"
      >
        Steak
      </p>
      <p
        className="foodPrice"
      >
        £18.00
      </p>
    </div>
    <div
      className="foodOptionBar"
      onClick={[Function]}
    >
      <p
        className="foodName"
      >
        Meatballs
      </p>
      <p
        className="foodPrice"
      >
        £11.50
      </p>
    </div>
    <div
      className="foodOptionBar"
      onClick={[Function]}
    >
      <p
        className="foodName"
      >
        Salmon fillet
      </p>
      <p
        className="foodPrice"
      >
        £14.00
      </p>
    </div>
    <div
      className="foodOptionBar"
      onClick={[Function]}
    >
      <p
        className="foodName"
      >
        Vegetarian lasagna
      </p>
      <p
        className="foodPrice"
      >
        £12.00
      </p>
    </div>
  </div>
  <div
    className="courseMenu"
  >
    <h3>
      Desserts
    </h3>
    <div
      className="foodOptionBar"
      onClick={[Function]}
    >
      <p
        className="foodName"
      >
        Sticky toffee
      </p>
      <p
        className="foodPrice"
      >
        £18.00
      </p>
    </div>
    <div
      className="foodOptionBar"
      onClick={[Function]}
    >
      <p
        className="foodName"
      >
        Tiramisu
      </p>
      <p
        className="foodPrice"
      >
        £4.50
      </p>
    </div>
    <div
      className="foodOptionBar"
      onClick={[Function]}
    >
      <p
        className="foodName"
      >
        Cheesecake
      </p>
      <p
        className="foodPrice"
      >
        £4.00
      </p>
    </div>
    <div
      className="foodOptionBar"
      onClick={[Function]}
    >
      <p
        className="foodName"
      >
        Ice cream
      </p>
      <p
        className="foodPrice"
      >
        £3.50
      </p>
    </div>
  </div>
</div>
`;
